### PR TITLE
Cleanup update_if_exists vs skip_if_exists.

### PR DESCRIPTION
**Pull request by agorajek**
### Summary

I noticed that in the case of create_namespace() and create_tag() the argument `update_if_exists` was misleading. Adjusted it to `skip_if_exists` . The update if exist functionality is still possible through the get()/save() pattern.

Also found one more issue with the _deploy_namespaces() method of the project compile.
It was erroring out because of existing namespaces, example:
```
  users.olek.project1.xp.source-controlled     tag         Tag users.olek.project1.xp.source-controlled successfully created (or updated)
  users.olek.project1                          namespace   {'message': 'Node namespace `users.olek.project1` already exists'}
```

### Test Plan

Unit test plus test deployments of my example YAML project.

Now getting a warning if the namespace exists:
```
                                                                                                                      XP Nodes
                                                                                                     Deployment for Prefix: users.olek.project1

  Name                                         Type        Message
 ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
  users.olek.project1.xp.source-controlled     tag         Tag users.olek.project1.xp.source-controlled successfully created (or updated)
  users.olek.project1                          namespace   Namespace users.olek.project1 already exists
  users.olek.project1.xp                       namespace   Namespace users.olek.project1.xp already exists
  users.olek.project1.xp.foo-bar               dimension   Node users.olek.project1.foo-bar successfully created (or updated)
...
```

- [x] PR has an associated issue: #1192 
- [x] `make check` passes
- [x] `make test` shows 100% unit test coverage

### CODE REVIEW

File: datajunction-clients/python/datajunction/builder.py [hunk 1]
```diff
@@ -502,7 +502,7 @@ def create_tag(  # pylint: disable=too-many-arguments
         try:
             self._create_tag(tag=new_tag)
         except DJTagAlreadyExists as exc:
-            if not update_if_exists:
+            if not skip_if_exists:
```
Comment (shangyian) [inline datajunction-clients/python/datajunction/builder.py:505]: How come for tags it's this  flag? Like I could imagine a need to update the tag if it exists, say if the metadata json blob on the tag changes, or the display name changes

**Review by shangyian — APPROVED** [review]

File: datajunction-clients/python/datajunction/builder.py (hunk 1 above)
Comment (agorajek) [reply datajunction-clients/python/datajunction/builder.py:505]: You are right... and there is a way to do that through tag's `get` and `save` methods. Just these `create_<something>` had a different approach. It was just too much to fix in one PR.

File: datajunction-clients/python/datajunction/__about__.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
 """
 Version for Hatch
 """
-__version__ = "0.0.1a68"
+__version__ = "0.0.1a68.dev0"
```
Comment (agorajek) [inline datajunction-clients/python/datajunction/__about__.py:4]: oh shoot

File: datajunction-clients/python/datajunction/builder.py (hunk 1 above)
Comment (agorajek) [reply datajunction-clients/python/datajunction/builder.py]: It wasn't that bad after all. Changed `create_tag()` to update_if_exists.

**Pull request merged**